apis/v1alpha2/validation/httproute.go

```diff
@@ -64,7 +68,32 @@ func validateHTTPRouteUniqueFilters(rules []gatewayv1a2.HTTPRouteRule, path *fie
 			}
 		}
 
+		if errList := validateHttpRouteBackendRefs(rule.BackendRefs, path, i); len(errList) > 0 {
+			errs = append(errs, errList...)
+		}
 	}
 
 	return errs
 }
+
+func validateHttpRouteBackendRefs(ref []gatewayv1a2.HTTPBackendRef, path *field.Path, i int) field.ErrorList {
+	var errs field.ErrorList
+
+	for _, bkr := range ref {
+		counts := map[gatewayv1a2.HTTPRouteFilterType]int{}
+		for _, filter := range bkr.Filters {
+			counts[filter.Type]++
+		}
+
+		for _, key := range repeatableHTTPRouteFilters {
+			delete(counts, key)
+		}
+
+		for filterType, count := range counts {
+			if count > 1 {
+				errs = append(errs, field.Invalid(path.Index(i).Child("BackendRefs"), filterType, "cannot be used multiple times in the same rule"))
+			}
+		}
+	}
+	return errs
+}
```

Comment:
Please add tests for this function.

Reply:
ed56896aa89ed5bc26139ec27ebbbe6fd54c653e